### PR TITLE
vyper: use apeworx

### PR DESCRIPTION
**Pull request by yann300**
Update Vyper plugin to use the new compiler endpoint in docker from Apeworx

### CODE REVIEW

File: apps/remix-ide/src/app.js [hunk 1]
```diff
@@ -130,12 +131,10 @@ class AppComponent {
...
       '6fd22d6fe5549ad4c4d8fd3ca0b7816b.mod': 35 // remix desktop
     }
-    
-    this.matomoConfAlreadySet = Registry.getInstance().get('config').api.exists('settings/matomo-analytics')
-    this.matomoCurrentSetting = Registry.getInstance().get('config').api.get('settings/matomo-analytics')
-    this.showMatamo = matomoDomains[window.location.hostname] && !this.matomoConfAlreadySet
-    
-    this.walkthroughService = new WalkthroughService(appManager)
+    this.showMatamo = matomoDomains[window.location.hostname] && !Registry.getInstance().get('config').api.exists('settings/matomo-analytics')
+    this.showEnter = matomoDomains[window.location.hostname] && !localStorage.getItem('hadUsageTypeAsked')
```
Comment (LianaHus) [inline apps/remix-ide/src/app.js:135]: i think you have old code here @yann300

File: apps/vyper/src/app/utils/compiler.tsx [hunk 2]
```diff
@@ -30,48 +32,113 @@ export function isCompilationError(output: VyperCompilationOutput): output is Vy
   return output.status === 'failed'
 }
 
+export function normalizeContractPath(contractPath: string): string[] {
+  const paths = contractPath.split('/')
+  const filename = paths[paths.length - 1].split('.')[0]
+  let folders = ''
+  for (let i = 0; i < paths.length - 1; i++) {
+    if(i !== paths.length -1) {
```
Comment (LianaHus) [inline apps/vyper/src/app/utils/compiler.tsx:40]: space is missing after if

File: libs/remix-ui/solidity-compile-details/src/lib/solidity-compile-details.tsx [hunk 3]
```diff
@@ -4,73 +4,32 @@ import { TreeView, TreeViewItem } from '@remix-ui/tree-view'
...
+  selectedContract?: string
+  help?: any
+  insertValue?: any
   saveAs: any
 }
 
 const _paq = (window._paq = window._paq || [])
 
 export function RemixUiCompileDetails({ plugin, contractProperties, selectedContract, saveAs, help, insertValue }: RemixUiCompileDetailsProps) {
+  console.log(contractProperties)
```
Comment (LianaHus) [inline libs/remix-ui/solidity-compile-details/src/lib/solidity-compile-details.tsx:21]: need this?

File: libs/remix-ui/solidity-compiler/src/lib/contract-selection.tsx [hunk 4]
```diff
@@ -310,6 +310,7 @@ export const ContractSelection = (props: ContractSelectionProps) => {
                 </span>
               </CustomTooltip>
             </button>
+            {console.log(payload)}
```
Comment (LianaHus) [inline libs/remix-ui/solidity-compiler/src/lib/contract-selection.tsx:313]: remove?

File: libs/remix-ui/vyper-compile-details/src/lib/vyperCompile.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    [method: string]: string
+  }
+}
+
+export interface VyperCompileProps {
+  result: VyperCompilationResult
+  theme?: ThemeKeys | ThemeObject
+  themeStyle?: any
+}
+
```
Comment (LianaHus) [inline libs/remix-ui/vyper-compile-details/src/lib/vyperCompile.tsx:26]: extra line

File: apps/vyper/src/app/utils/compiler.tsx (hunk 2 above)
Comment (joeizang) [reply apps/vyper/src/app/utils/compiler.tsx:40]: fixed

File: libs/remix-ui/solidity-compile-details/src/lib/solidity-compile-details.tsx (hunk 3 above)
Comment (joeizang) [reply libs/remix-ui/solidity-compile-details/src/lib/solidity-compile-details.tsx:21]: No

File: libs/remix-ui/solidity-compiler/src/lib/contract-selection.tsx (hunk 4 above)
Comment (joeizang) [reply libs/remix-ui/solidity-compiler/src/lib/contract-selection.tsx:313]: no

File: libs/remix-ui/vyper-compile-details/src/lib/vyperCompile.tsx (hunk 5 above)
Comment (joeizang) [reply libs/remix-ui/vyper-compile-details/src/lib/vyperCompile.tsx:26]: removed

File: apps/remix-ide/src/app/plugins/vyper-compilation-details.tsx [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+const profile = {
+  name: 'vyperCompilationDetails',
+  displayName: 'Vyper Compile Details',
+  description: 'Displays details from vyper compiler',
+  location: 'mainPanel',
+  methods: ['showDetails'],
+  events: []
+}
+
+export class VyperCompilationDetailsPlugin extends ViewPlugin {
```
Comment (yann300) [inline apps/remix-ide/src/app/plugins/vyper-compilation-details.tsx:19]: This looks a bit strange that we aren't re using the solidity comp details plugin

Comment (joeizang) [reply apps/remix-ide/src/app/plugins/vyper-compilation-details.tsx:19]: reusing the solidity comp details is the ultimate plan. Sadly this hasn't been merged in time.

**Review by bunsenstraat — APPROVED** [review]

**Pull request merged**